### ISSUE TITLE
invalid-name warnings do not distinguish true constants

### ISSUE THREAD

**Issue by bje-** [PyCQA/pylint#5677]
### Bug description

"""very minimal test case"""

x = 10
x += 1

### Command used

```shell
pylint testcase.py
```

### Pylint output

```shell
testcase.py:3: [C0103(invalid-name), ] Constant name "x" doesn't conform to '(([A-Z_][A-Z0-9_]*)|(__.*__))$' pattern
```

### Expected behavior

No output is expected. In this example, `x` is not a constant. Data flow analysis can distinguish between global variables that are initialised only once (as constants) and global variables that are modified throughout the script. It is misleading for code readers if Pylint encourages the variable name to be put into uppercase if it is not a truly constant value.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0]
```

### OS / Environment

Ubuntu 20.04

**Comment by Pierre-Sassoulas** [comment]
This kind of is a duplicate of other issue regarding ``invalid-name``, the proposition was that both ``snake_case`` and ``UPPERCASE_WITH_UNDERSCORE`` should be permissible as there is no sane/fast way to check if a module level variable is a constant. This is probably a high priority issue seeing the number of duplicate that exists. (https://github.com/PyCQA/pylint/issues/4606, https://github.com/PyCQA/pylint/issues/3585)

**Comment by Pierre-Sassoulas** [comment]
Closing as a duplicate of #3585
